lib/Fhp/MT940/MT940.php

```diff
@@ -92,19 +92,35 @@ public function parse(string $rawData): array
                     // 16 = year
                     // 0511 = valuta date
                     // 0509 = booking date
+                    
                     $year = substr($transaction, 0, 2);
                     $valutaDate = $this->getDate($year . substr($transaction, 2, 4));
 
-                    $bookingDate = substr($transaction, 6, 4);
-                    if (preg_match('/^\d{4}$/', $bookingDate)) {
-                        // if valuta date is earlier than booking date, then it must be in the new year.
-                        $year = substr($transaction, 2, 2) < substr($transaction, 6, 2) ? --$year : $year;
-                        if (substr($transaction, 2, 2) == '12' && substr($transaction, 6, 2) == '01') {
-                            ++$year;
-                        } elseif (substr($transaction, 2, 2) == '01' && substr($transaction, 6, 2) == '12') {
-                            --$year;
+                    if (preg_match('/^\d{4}$/', substr($transaction, 6, 4)) === 1) {
+                        // try to guess the correct year of the booking date
+
+                        $valutaDateTime = new \DateTime($valutaDate);
+                        $bookingDateTime = new \DateTime($this->getDate($year . substr($transaction, 6, 4)));
```

Comment:
Ich würde noch

`$bookingDate = substr($transaction, 6, 4);´

wieder einführen, substr($transaction, 6, 4) kommt mehrfach vor.

Reply:
Ja da hast du Recht, ich habe es jetzt $bookingDatePart genannt weil $bookingDate ja schon in anderer Bedeutung verwendet wird.